### PR TITLE
Enable Osmosis chain configuration overwrite

### LINKED ISSUE #809: Expose configuration as environment variables
Currently, the process to connect the frontend to a new testnet is:

- create a new PR 
- have Vercel deploy it
- adapt the code to make the required changes (change `RPC`, `LCD`, `chain-id`, etc...)

However, this process is not optimal, and it tends to be repeated multiple times:

image

A better approach would be to generalize the changes made in this PR https://github.com/osmosis-labs/osmosis-frontend/pull/793, allowing users to configure the frontend using environment variables as input. The environment variable should be used **at RUNTIME** and not during the **BUILD**.

By doing so, we can quickly create different environments specifying only different environment variables and possibly use a different deployment solution than Vercel.

### PR DESCRIPTION
**Pull request by daniel-farina**
This PR will close #809. 

- Allows overwriting Osmosis chain config with custom parameters set on a .env configuration file. They are only overwriten when present.

### CODE REVIEW

**Comment by niccoloraspa** [comment]
Are the environment variables used at build time or runtime?

**Comment by daniel-farina** [comment]
> Are the environment variables used at build time or runtime?

I believe at build time but maybe @jonator  can confirm. On dev, they don't get applied unless I stop and start the dev server.

File: packages/web/README.md [hunk 1]
```diff
@@ -45,3 +45,7 @@ On completion, two local html files containing visual bundle trees for client an
 ### Frontier
 
 Frontier mode is managed by the `NEXT_PUBLIC_IS_FRONTIER=true` env var, deployed from `master` branch.
+
+## Environment Variables
+
```
Comment (jonator) [inline packages/web/README.md:50]: Could we add one sentence of how the config works? Perhaps replace everything in this section with:

> By default, chain configuration is hardcoded and determined by the `NEXT_PUBLIC_IS_TESTNET` env var for developer convenience and simplicity. Please directly change those values should the config be changed from here on. For temporary overrides, consult the [.env](.env) file.

File: packages/web/.env [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# More information about how this works [here](https://nextjs.org/docs/basic-features/environment-variables).
+
+# Osmosis Chain Configuration Overwrite
+# NEXT_PUBLIC_IS_FRONTIER=false
+# NEXT_PUBLIC_IS_TESTNET=true
+# NEXT_PUBLIC_OSMOSIS_RPC_OVERWRITE=https://rpc-test.osmosis.zone/
+# NEXT_PUBLIC_OSMOSIS_REST_OVERWRITE=https://lcd-test.osmosis.zone/
+# NEXT_PUBLIC_OSMOSIS_CHAIN_ID_OVERWRITE=osmo-test-4
+# NEXT_PUBLIC_OSMOSIS_EXPLORER_URL_OVERWRITE=https://testnet.mintscan.io/osmosis-testnet/txs/{txHash}
+# NEXT_PUBLIC_OSMOSIS_CHAIN_NAME_OVERWRITE=Osmosis (Testnet)
```
Comment (jonator) [inline packages/web/.env:12]: I'm a fan of these commented vars as example!

Comment (jonator) [inline packages/web/.env:12]: I'm a fan of these commented vars as an example! Makes things really easy.

File: packages/web/config/chain-infos.ts [hunk 3]
```diff
@@ -3,18 +3,25 @@ import { Bech32Address } from "@keplr-wallet/cosmos";
 import { createKeplrChainInfos, SimplifiedChainInfo } from "./utils";
 
 const IS_TESTNET = process.env.NEXT_PUBLIC_IS_TESTNET === "true";
+const RPC_OVERWRITE = process.env.NEXT_PUBLIC_RPC_OVERWRITE;
```
Comment (jonator) [inline packages/web/config/chain-infos.ts:6]: Could we prepend `OSMOSIS_` to each of these constants to clarify usage?

File: packages/web/config/chain-infos.ts [hunk 4]
```diff
@@ -3,18 +3,25 @@ import { Bech32Address } from "@keplr-wallet/cosmos";
...
+const REST_OVERWRITE = process.env.NEXT_PUBLIC_REST_OVERWRITE;
+const EXPLORER_UR_OVERWRITE = process.env.NEXT_PUBLIC_EXPLORER_URL_OVERWRITE;
```
Comment (jonator) [inline packages/web/config/chain-infos.ts:8]: Typo at `UR`->`URL`.

**Review by jonator — CHANGES_REQUESTED** [review]

**Review by ankurdotb — CHANGES_REQUESTED** [review]

**Comment by jonator** [comment]
Closes #809

**Comment by jonator** [comment]
@daniel-farina still want this merged, or is it stale?

File: packages/web/config/chain-infos.ts (hunk 3 above)
Comment (daniel-farina) [reply packages/web/config/chain-infos.ts]: done~

File: packages/web/config/chain-infos.ts (hunk 4 above)
Comment (daniel-farina) [reply packages/web/config/chain-infos.ts]: fixed

**Comment by daniel-farina** [comment]
This is ready ser @jonator  @niccoloraspa 
Ready to test on stage.

**Comment by jonator** [comment]
Please resolve conflicts @daniel-farina

**Comment by daniel-farina** [comment]
> Please resolve conflicts @daniel-farina

Conflicts fixed.

**Review by jonator — APPROVED** [review]

**Pull request merged**